src/Agent.Listener/Configuration/LinuxServiceControlManager.cs

```diff
@@ -253,8 +254,10 @@ private void ExecuteSystemdCommand(string command)
                 var processInvoker = HostContext.CreateService<IProcessInvoker>();
 
                 // TODO: can systemd installed in non default directory?
-                processInvoker.Execute("/usr/bin", "systemctl", command, null);
-                processInvoker.WaitForExit(HostContext.CancellationToken);
+                using (var cs = new CancellationTokenSource(TimeSpan.FromSeconds(45)))
+                {
+                    processInvoker.ExecuteAsync("/usr/bin", "systemctl", command, null, cs.Token).Wait();
```

Comment:
I agree that Wait should be used only at the OS entry point function. If I follow up, I need to make ExecuteSystemdCommand async, which brings other changes. I'm afraid of merge conflicts.  Maybe @kasubram can improve this code.